File: schemaorggo/reserveaction.go
package schemaorggo

import "encoding/json"

// ReserveAction see : https://schema.org/ReserveAction
type ReserveAction struct {
	PlanAction

	typeContext

	// ScheduledTime see : https://schema.org/scheduledTime
	// The time the object is scheduled to.
	ScheduledTime DateTime `json:"scheduledTime,omitempty"`
}

func (v ReserveAction) MarshalJSONWithTypeContext() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "ReserveAction"

	return json.Marshal(v)
}

func (v *ReserveAction) MarshalJSON() ([]byte, error) {
	if v == nil {
		return []byte("null"), nil
	}

	v.C = "http://schema.org"
	v.T = "ReserveAction"

	return json.Marshal(*v)
}

File: schemaorgphp/classes/reserve_action.php
<?php

class ReserveAction extends PlanAction implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'ReserveAction';
	
	/**
	 * The time the object is scheduled to.
	 * see : https://schema.org/scheduledTime
	 * @var string|string[]
	 */
	public var $scheduled_time;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'ReserveAction'
		);
		
		$serialized = so_json_serialize( $this->scheduled_time );
		if ( ! empty( $serialized ) ) {
			$out['scheduledTime'] = $serialized;
		}
		
		return $out;
	}
}
